master/internal/configpolicy/task_config_policy_intg_test.go

```diff
@@ -46,9 +49,144 @@ func TestPriorityAllowed(t *testing.T) {
 	require.False(t, ok)
 }
 
-func addWorkspacePriorityLimit(t *testing.T, pgDB *db.PgDB, limit int) model.Workspace {
-	ctx := context.Background()
+func TestValidateNTSCConstraints(t *testing.T) {
+	require.NoError(t, etc.SetRootPath(db.RootFromDB))
+	pgDB, cleanup := db.MustResolveNewPostgresDatabase(t)
+	defer cleanup()
+	db.MustMigrateTestPostgres(t, pgDB, db.MigrationsFromDB)
+
+	wkspPriorityLimit := 7
+	user := db.RequireMockUser(t, pgDB)
+
+	t.Run("no constraints set - ok", func(t *testing.T) {
+		resourceManager := mocks.ResourceManager{}
+		resourceManager.On("SmallerValueIsHigherPriority", mock.Anything).Return(false, nil)
+
+		config := defaultConfig()
+		ok, err := ValidateNTSCConstraints(context.Background(), 1, config, &resourceManager)
+		require.NoError(t, err)
+		require.True(t, ok)
+	})
+
+	t.Run("running in wksp with constraints - not ok", func(t *testing.T) {
+		w := addWorkspacePriorityLimit(t, user, wkspPriorityLimit)
+		resourceManager := mocks.ResourceManager{}
+		resourceManager.On("SmallerValueIsHigherPriority", mock.Anything).Return(false, nil)
+
+		config := defaultConfig()
+		ok, err := ValidateNTSCConstraints(context.Background(), w.ID, config, &resourceManager)
+		require.False(t, ok)
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "requested priority")
+	})
+
+	t.Run("running in wksp without constraints - ok", func(t *testing.T) {
+		resourceManager := mocks.ResourceManager{}
+		resourceManager.On("SmallerValueIsHigherPriority", mock.Anything).Return(false, nil)
+		w := model.Workspace{Name: uuid.NewString(), UserID: user.ID}
+		_, err := db.Bun().NewInsert().Model(&w).Exec(context.Background())
+		require.NoError(t, err)
+
+		config := defaultConfig()
+		ok, err := ValidateNTSCConstraints(context.Background(), w.ID, config, &resourceManager)
+		require.True(t, ok)
+		require.NoError(t, err)
+	})
+
+	t.Run("exceeds max slots - not ok", func(t *testing.T) {
+		constraints := DefaultConstraints()
+		w := model.Workspace{Name: uuid.NewString(), UserID: user.ID}
+		_, err := db.Bun().NewInsert().Model(&w).Exec(context.Background())
+		require.NoError(t, err)
+		addConstraints(t, user, &w.ID, *constraints)
+
+		resourceManager := mocks.ResourceManager{}
+		resourceManager.On("SmallerValueIsHigherPriority", mock.Anything).Return(true, nil)
+
+		config := defaultConfig()
+		ok, err := ValidateNTSCConstraints(context.Background(), w.ID, config, &resourceManager)
+		require.False(t, ok)
+		require.Error(t, err)
```

Comment:
Super duper nit: huh, this is interesting! I'm accustomed to the idiom of assuming nothing about other returned values when a non-nil error is returned. Do we perhaps want to skip this `require.False(t, ok)` check?